scripts/gendeclarations.lua

```diff
@@ -87,6 +87,15 @@ local spaces = any_amount(branch(
       any_character
     )),
     lit('\n')
+  ),
+  -- Linemarker inserted by preprocessor
+  concat(
+    lit('# '),
```

Comment:
The only thing that resembles a preprocessor directive are the linemarkers in the processed `*.i` files (see build/src/nvim/auto/).
But if you are concerned about constructs like `#...`, I also have another fix (https://github.com/oni-link/neovim/commit/f96e9b27498ed8360642988a0d20d409fa0f18e4) that tells the preprocessor to generate the `*.i` files without any `#...`. This fix is simpler and faster, but it generates some additional declarations that were (already) included from other header files. Additional generated declarations for `screen.c` for example are:
```c
static __inline unsigned int __bswap_32(unsigned int __bsx);
static __inline __uint64_t __bswap_64(__uint64_t __bsx);
static inline void *ga_append_via_ptr(garray_T *gap, size_t item_size);
static inline _Bool ascii_iswhite(int c);
static inline _Bool ascii_isdigit(int c);
static inline _Bool ascii_isxdigit(int c);
static inline _Bool ascii_isspace(int c);
static inline _Bool arabic_char(int c);
```